Classes/Controller/BackendServiceController.php

```diff
@@ -176,33 +184,41 @@ public function changeAction(array $changes): void
      */
     public function publishChangesInSiteAction(array $command): void
     {
-        /** @todo send from UI */
-        $contentRepositoryId = SiteDetectionResult::fromRequest($this->request->getHttpRequest())->contentRepositoryId;
-        $command['contentRepositoryId'] = $contentRepositoryId->value;
-        $command['siteId'] = $this->nodeService->deserializeNodeAddress(
-            $command['siteId'],
-            $contentRepositoryId
-        )->nodeAggregateId->value;
-        $command = PublishChangesInSite::fromArray($command);
-
         try {
+            /** @todo send from UI */
+            $contentRepositoryId = SiteDetectionResult::fromRequest($this->request->getHttpRequest())->contentRepositoryId;
+            $command['contentRepositoryId'] = $contentRepositoryId->value;
+            $command['siteId'] = $this->nodeService->deserializeNodeAddress(
+                $command['siteId'],
+                $contentRepositoryId
+            )->nodeAggregateId->value;
+            $command = PublishChangesInSite::fromArray($command);
             $workspace = $this->workspaceProvider->provideForWorkspaceName(
                 $command->contentRepositoryId,
                 $command->workspaceName
             );
-            $workspace->publishChangesInSite($command->siteId);
-
-            $success = new Success();
-            $success->setMessage(sprintf('Published.'));
-
-            $this->feedbackCollection->add($success);
+            $publishingResult = $workspace
+                ->publishChangesInSite($command->siteId);
+
+            $contentRepository = $this->contentRepositoryRegistry->get($contentRepositoryId);
+            $baseWorkspaceName = $contentRepository->getWorkspaceFinder()->findOneByName($command->workspaceName)->baseWorkspaceName;
```

Comment:
Good news: the Workspace model has a getCurrentBaseWorkspace() method for this

Reply:
Thanks for the pointer! Adjusted.